pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -121,6 +122,15 @@ type OrchestrationStatus struct {
 
 // Start the osd management
 func (c *Cluster) Start() error {
+	// Validate PODs memory if specified
+	// That's the best we can do, hopefully the node has enough RAM to run the process
+	// Further check should be implemented in the ceph daemon
+	// This is valid for both Filestore and Bluestore
+	osdPodMemoryLimit := c.resources.Limits.Memory().Value() // Bytes
```

Comment:
It seems like this ~4 line pattern is repeated for all/most of the daemons. This seems like it might be a good candidate for pulling into a helper method. Into `pkg/operator/ceph/spec` perhaps?